cpp/ql/lib/semmle/code/cpp/ir/implementation/raw/internal/IRConstruction.qll

```diff
@@ -386,6 +386,16 @@ Instruction getPrimaryInstructionForSideEffect(SideEffectInstruction instruction
         .getPrimaryInstructionForSideEffect(getInstructionTag(instruction))
 }
 
+predicate hasUnreachedInstruction(IRFunction func) {
+  exists(Call c | 
+    c.getEnclosingFunction() = func.getFunction() and
+    (
+      c.getTarget().hasSpecifier("_Noreturn") or
```

Comment:
As far as I remember `_Noreturn` is represented as `noreturn` in the database.

Wouldn't it be better to use the `exits(Function f)` predicate from `DefaultOptions.qll`? And, `exprExits(Expr e)` if you want to do something similar for `assume(0)` expressions?

Reply:
For reference: https://github.com/github/codeql/commit/1a1c34e1bebf558142af6305c144a88345059714